lib/encoder.js

```diff
@@ -752,6 +755,31 @@ function defineInstanceMembers() {
     parts.unshift(this.getLengthBuffer(propCounter));
     return Buffer.concat(parts);
   };
+  this.encodeJson = function encodeJson(value){
+    let jsonString;
+    //Use a switch to handle other typeof checking in the future
+    switch(typeof value){
+      case 'string':
+        // If valid json was already provided (but stringified) use the provided string
+        // This is valid if a user wants absolute control over the json object they pass for storage
+        // Passing an object as a value would leave some risk of the object being modified by the stack after client.execute() is called
+        try{
```

Comment:
This try/catch block should actually be removed so malformed json strings do not make it past this point.

Reply:
The code in the try/catch statement should still stay though. This will just allow JSON.parse to throw if the string is malformed (just like the JSON.stringify would in the default: switch statement)